cuML/src/glm/qn/simple_mat.h

```diff
@@ -294,55 +294,48 @@ std::ostream &operator<<(std::ostream &os, const SimpleMat<T> &mat) {
   return os;
 }
 
-// TODO The following classes own their storage and are just for testing
-// The internals should be replaced by device_buffer+device_allocator, when ready
 template <typename T> struct SimpleVecOwning : SimpleVec<T> {
   typedef SimpleVec<T> Super;
-  bool allocated;
+  typedef MLCommon::device_buffer<T> Buffer;
+  std::unique_ptr<Buffer> buf;
 
-  SimpleVecOwning() : Super(), allocated(false) {}
+  SimpleVecOwning() : Super() {}
 
-  SimpleVecOwning(int n) : Super(), allocated(false) { reset(n); }
-
-  ~SimpleVecOwning() noexcept(false) { CUDA_CHECK(cudaFree(Super::data)); }
+  SimpleVecOwning(const cumlHandle_impl &handle, int n, cudaStream_t stream)
```

Comment:
Why not only pass in a `std::shared_ptr<device_allocator>` instead of `cumlHandle_impl`?